src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -262,16 +263,16 @@ async def _exec(
         run_id = run_id or str(uuid.uuid4())
 
         # execute lines
+        is_timeout = False
         if isinstance(self._executor_proxy, PythonExecutorProxy):
-            line_results.extend(
-                self._executor_proxy._exec_batch(
-                    batch_inputs,
-                    output_dir,
-                    run_id,
-                    batch_timeout_sec=self._batch_timeout_sec,
-                    line_timeout_sec=self._line_timeout_sec,
-                )
+            results, is_timeout = self._executor_proxy._exec_batch(
+                batch_inputs,
+                output_dir,
+                run_id,
+                batch_timeout_sec=self._batch_timeout_sec,
+                line_timeout_sec=self._line_timeout_sec,
             )
+            line_results.extend(results)
         else:
             await self._exec_batch(line_results, batch_inputs, run_id)
```

Comment:
Add TODO about batch_timeout for this code path,

Reply:
Added.